### ISSUE TITLE
Tenancy Term and Break Clause details are not available in any of the Views from Reapit to Snowflake. can you help me how can we  get the BreakClause details to work on our Lettings Revenue reports. 

### ISSUE THREAD

**Issue by SairamKDXFT** [reapit/foundations#9857]
**Background context**

**Specification**
Tenancy Term and Break Clause details are not available in any of the Views from Reapit to Snowflake. can someone help me how can we get the Break Clause details to work on our Lettings Revenue reports.

**Comment by plittlewood-rpt** [comment]
This will need properly scoping out to look at the best way of exposing break clause information in the analytics schema

**Comment by plittlewood-rpt** [comment]
HI @SairamKDXFT I cannot give a date at the moment as we need to spec out what this data will look like in the analytics schema. As soon as it's been scoped out properly, this ticket will be updated. Thanks

**Comment by SairamKDXFT** [comment]
@plittlewood-rpt @HollyJoyPhillips  may i know when this can be available in analytics Schema.

**Comment by HollyJoyPhillips** [comment]
Hi @SairamKDXFT, unfortunately we don't currently have a timeframe for adding this to the schema as it's with our Foundations team for prioritisation. 

We will update the ticket with more information where possible.

**Comment by SairamKDXFT** [comment]
Hi @plittlewood-rpt @HollyJoyPhillips , Could we schedule a call to discuss how we can calculate revenue from the existing data in Snowflake? This is urgent, and we need the revenue information as soon as possible.

**Comment by SairamKDXFT** [comment]
Hi @plittlewood-rpt @HollyJoyPhillips any response on this will be really helpful, we are having trouble in accessing the Revenue for Lettings.

**Comment by HollyJoyPhillips** [comment]
Hi @SairamKDXFT sorry for the delay, let me review with the team and come back to as soon as possible. Thanks.

**Comment by plittlewood-rpt** [comment]
Hi @SairamKDXFT whilst we look at this, it would be useful if you can elaborate a bit on what you actually need to satisfy your reporting requirement. "Tenancy Term and Break Clause details" is a little vague so anything you can give us to help refine the request would be helpful. Thank you

**Comment by SairamKDXFT** [comment]
@plittlewood-rpt  on a Tenancy record level, we need to get the revenue/ Fee receivable.  

- till the break clause
- post the break clause. 
- Renewal fee as per the changes (if any). 

also is there any procedure to get the track the Lettings Monthly revenue of the Branch, Lettings  Monthly Negotiator earnings.

**Comment by HollyJoyPhillips** [comment]
Hi @SairamKDXFT, would you be able to provide screenshots/examples of the fields within the CRM that those items relate to please?

**Comment by SairamKDXFT** [comment]
@HollyJoyPhillips Currently, in our CRM system, we don't have detailed Lettings Fee/Revenue information at the same granular level as we do for Sales Property.
For each tenancy we need Revenue/Fee receivable for company. 

In CRM we have these numbers under 'Tenancy Income' and it's not coming to Snowflake.

**Comment by HollyJoyPhillips** [comment]
Hi @SairamKDXFT thank you. Are you able to provide a screenshot (with PII removed) of the field location please?

**Comment by SairamKDXFT** [comment]
@HollyJoyPhillips   could can find the details in Accounts function section  ( Bottom right corner ) on the Tenancy Screen.  
<img width="211" alt="image" src="https://github.com/reapit/foundations/assets/145762206/84d85fd3-af83-485e-980f-c7739f2d5360">


@HollyJoyPhillips ; this is so time consuming; can we get into a call to discuss further ?

**Comment by plittlewood-rpt** [comment]
HI @SairamKDXFT I'm trying to understand this ticket a little better. Your summary mentions the tenancy term and break clause details, but the comments seem to relate specifically to the tenancy income report that's been recently added to the CRM. Can you just confirm which it is you're after here? Thanks

**Comment by SairamKDXFT** [comment]
@plittlewood-rpt  this was raised on  27th September 2023, to include the BreakClause details in the analytics Schema. 

in the conversation on this ticket  https://github.com/reapit/foundations/issues/9857#issuecomment-1988182078 ; was asked about what actually need to satisfy my reporting requirement.    

We need to build a lettings Revenue report on a Tenancy record level with the data from the snowflake. 
- till the break clause
- post the break clause to End of tenure.
- Renewal fee as per the changes (if any).

can you advise how this can be achieved with the available data.

**Comment by SairamKDXFT** [comment]
@HollyJoyPhillips @plittlewood-rpt   If these details were included in the analytical schema, it would be advantageous. 

However, if they are not readily available, could you please advise on the procedure for calculating the same data using the available fields from Snowflake.

**Comment by HollyJoyPhillips** [comment]
Hi @SairamKDXFT thank you for confirming. I've reached out to our product team and will reply as soon as possible. Thanks.

**Comment by HollyJoyPhillips** [comment]
Product have approved surfacing the above information. We will move this back into refinement for a specification to to be added.

**Comment by plittlewood-rpt** [comment]
To discussed internally with MM about this.

**Comment by SairamKDXFT** [comment]
@plittlewood-rpt yes, we need Break Clause for both Tenancies and TENANCY_RENEWALS. 

our business team requires this data to calculate Revenue from Lettings and Renewals up to the Break Clause. 

or could you please help me understand how we can obtain this from existing Data.

**Comment by plittlewood-rpt** [comment]
Hi @SairamKDXFT sorry for the delay in responding. It feels like we are muddying the water a little here as we've been talking about the tenancy income report but I'm not sure if what you actually want is the raw break clause information so you can just calculate this yourself. Or indeed is it both (in which case I need to split this ticket into two). Can you let me know?
